tests/sources/test_vector4_impl.h

```diff
@@ -639,64 +629,47 @@ void test_vector4_impl(const FloatType threshold)
-	results[(int)comp0][(int)comp1][(int)comp2][(int)mix4::c] = vector_all_near_equal(vector_mix<comp0, comp1, comp2, mix4::c>(input0, input1), scalar_mix<Vector4Type, comp0, comp1, comp2, mix4::c>(input0, input1), threshold); \
-	results[(int)comp0][(int)comp1][(int)comp2][(int)mix4::d] = vector_all_near_equal(vector_mix<comp0, comp1, comp2, mix4::d>(input0, input1), scalar_mix<Vector4Type, comp0, comp1, comp2, mix4::d>(input0, input1), threshold)
-
-#define RTM_TEST_MIX_XY(results, input0, input1, comp0, comp1) \
-	RTM_TEST_MIX_XYZ(results, input0, input1, comp0, comp1, mix4::x); \
-	RTM_TEST_MIX_XYZ(results, input0, input1, comp0, comp1, mix4::y); \
-	RTM_TEST_MIX_XYZ(results, input0, input1, comp0, comp1, mix4::z); \
-	RTM_TEST_MIX_XYZ(results, input0, input1, comp0, comp1, mix4::w); \
-	RTM_TEST_MIX_XYZ(results, input0, input1, comp0, comp1, mix4::a); \
-	RTM_TEST_MIX_XYZ(results, input0, input1, comp0, comp1, mix4::b); \
-	RTM_TEST_MIX_XYZ(results, input0, input1, comp0, comp1, mix4::c); \
-	RTM_TEST_MIX_XYZ(results, input0, input1, comp0, comp1, mix4::d)
-
-#define RTM_TEST_MIX_X(results, input0, input1, comp0) \
-	RTM_TEST_MIX_XY(results, input0, input1, comp0, mix4::x); \
-	RTM_TEST_MIX_XY(results, input0, input1, comp0, mix4::y); \
-	RTM_TEST_MIX_XY(results, input0, input1, comp0, mix4::z); \
-	RTM_TEST_MIX_XY(results, input0, input1, comp0, mix4::w); \
-	RTM_TEST_MIX_XY(results, input0, input1, comp0, mix4::a); \
-	RTM_TEST_MIX_XY(results, input0, input1, comp0, mix4::b); \
-	RTM_TEST_MIX_XY(results, input0, input1, comp0, mix4::c); \
-	RTM_TEST_MIX_XY(results, input0, input1, comp0, mix4::d)
-
-	// This generates 8*8*8*8 = 4096 unit tests... it takes a while to compile and uses a lot of stack space
-	// Disabled by default to reduce the build time
-	bool vector_mix_results[8][8][8][8];
-
 	const FloatType test_value0_flt[4] = { FloatType(2.0), FloatType(9.34), FloatType(-54.12), FloatType(6000.0) };
 	const FloatType test_value1_flt[4] = { FloatType(0.75), FloatType(-4.52), FloatType(44.68), FloatType(-54225.0) };
 	const Vector4Type test_value0 = vector_set(test_value0_flt[0], test_value0_flt[1], test_value0_flt[2], test_value0_flt[3]);
 	const Vector4Type test_value1 = vector_set(test_value1_flt[0], test_value1_flt[1], test_value1_flt[2], test_value1_flt[3]);
 
-	RTM_TEST_MIX_X(vector_mix_results, test_value0, test_value1, XArg);
-
-	const int comp0 = (int)XArg;
-	for (int comp1 = 0; comp1 < 8; ++comp1)
-	{
-		for (int comp2 = 0; comp2 < 8; ++comp2)
-		{
-			for (int comp3 = 0; comp3 < 8; ++comp3)
-			{
-				INFO("vector_mix<" << comp0 << ", " << comp1 << ", " << comp2 << ", " << comp3 << ">");
-				REQUIRE(vector_mix_results[comp0][comp1][comp2][comp3] == true);
-			}
-		}
-	}
-
-#undef RTM_TEST_MIX_X
-#undef RTM_TEST_MIX_XY
-#undef RTM_TEST_MIX_XYZ
+	REQUIRE(scalar_near_equal(vector_get_x(vector_mix<mix4::x, mix4::a, mix4::a, mix4::a>(test_value0, test_value1)), test_value0_flt[0], threshold));
```

Comment:
This is not equivalent to the previous code. Why was it changed?
The previous code tested all possible permutations because various patterns optimize to different intrinsics on different platforms. Here by repeating the same members over and over, such patterns might not be tested.
The exhaustive nature ensures that the patterns are properly detected, optimized, and tested and that no other combination triggers the pattern incorrectly.
See the SSE impl of vector_mix for vector4f.

Reply:
Arg, sorry, you're right. I didn't look at vector_mix() and assumed it had a single possible implementation per compile environment (the fact it takes template params should have been a red flag...)

The problem with splitting the cpp file further was the AppVeyor builds started taking longer than an hour, and they just timed out and failed.

I've taken another crack at making the compilation faster, but this time running every possible permutation. We'll see how it goes :/